### ISSUE TITLE
Filter Communities by Enabled Community Stake

### ISSUE THREAD

**Issue by HIM92** [hicommonwealth/commonwealth#6692]
## Description

Goal: Users are able to filter communities by whether they have stake enabled or not

User story: As a user exploring communities, I want to filter out those without Community Stake enabled so that I can explore those communities and join them if they interest me 

## Project Owner
PM: @HIM92 @sachben91 
Designer: @jessmart1213 

##  Feature specs 
- Clicking the STAKE filter button on the Explore page should narrow down the list of communities to only show Stake-enabled communities.

<img width="414" alt="Screenshot 2024-02-22 at 1 25 09 PM" src="https://github.com/hicommonwealth/commonwealth/assets/153562904/d1b3af89-de64-41f3-af42-2f970411a5e0">

- The communities should be displayed sorting in descending order of number of users within a stake enabled community so that the bigger communities with stake are displayed first 
- The community cards should be updated to the new community card design, which is the same component that was used for community cards in the related communities page 

## Acceptance Criteria
- As a user exploring communities, I want to filter out those without Community Stake enabled by clicking on the STAKE filter in the explore page 
- The user should view the new community card design 

## Engineering requirements
- [x] Update filter row UI
- [x] Add stake filter 
- [x] Sort community by stake and member count

## Designs 
https://www.figma.com/file/omZZrndm6Phbk1rOU4IKhS/Stake-V1-Fast-Follows?type=design&mode=design&t=DkQY7hY46jMASDad-0

## Additional Context
- Other functionality on the page on a separate ticket 1) add stake tag into the filter row 2) add Join button onto each card

**Comment by HIM92** [comment]
AI @HIM92 define activity

**Comment by jnaviask** [comment]
TBD: need formal defn of activity.

**Comment by HIM92** [comment]
AI @HIM92 add figma links

**Comment by mzparacha** [comment]
Blocked by https://github.com/hicommonwealth/commonwealth/issues/6869
